crux-lucene/test/crux/lucene/extension_test.clj

```diff
@@ -22,5 +32,253 @@
+  (CustomEavIndexer. index-store))
+
+(defn ^Analyzer ->custom-analyzer
+  "case-insensitive analyzer"
+  [_]
+  (.build (doto (CustomAnalyzer/builder)
+            (.withTokenizer ^String KeywordTokenizerFactory/NAME ^"[Ljava.lang.String;" (into-array String []))
+            (.addTokenFilter ^String LowerCaseFilterFactory/NAME ^"[Ljava.lang.String;" (into-array String [])))))
+
+(defn ^Analyzer ->per-field-analyzer
+  "global analyzer to cover indexing and _nearly_ all queries (it can't help with overriding dynamic fields)"
+  [_]
+  (PerFieldAnalyzerWrapper. (->custom-analyzer nil) {field-crux-val-exact (KeywordAnalyzer.)}))
+
+(defn build-custom-query [analyzer field-string v]
+  (let [qp (doto (QueryParser. field-string analyzer) (.setAllowLeadingWildcard true))
+        b (doto (BooleanQuery$Builder.)
+            (.add (.parse qp v) BooleanClause$Occur/MUST))]
+    (.build b)))
+
+(defn custom-text-search
+  ([node db attr query]
+   (custom-text-search node db attr query {}))
+  ([node db attr query opts]
+   (with-open [s (l/search node query)]
+     (->> (iterator-seq s)
+          (map (fn doc->rel [[^Document doc score]]
+                 [(.get ^Document doc l/field-crux-val) score]))
+          (mapcat (fn resolve-atemporal-av [[v s]]
+                    (c/q db {:find '[e v s]
+                             :in '[v s]
+                             :where [['e attr 'v]]}
+                         v s)))
+          (into [])))))
+
+(defn custom-wildcard-text-search
+  ([node db query]
+   (custom-wildcard-text-search node db query {}))
+  ([node db query opts]
+   (with-open [s (l/search node
+                           query
+                           opts)]
+     (->> (iterator-seq s)
+          (map (fn doc->rel [[^Document doc score]]
+                 [(.get ^Document doc l/field-crux-attr) (.get ^Document doc l/field-crux-val) score]))
+          (mapcat (fn resolve-atemporal-av [[a v s]]
+                    (let [a (keyword a)]
+                      (c/q db {:find '[e a v s]
+                               :in '[a v s]
+                               :where [['e a 'v]]}
+                           a v s))))
+          (into [])))))
+
+(t/deftest test-custom-index-and-analyzer
+  (with-open [node (c/start-node {:crux.lucene/lucene-store {:analyzer 'crux.lucene.extension-test/->per-field-analyzer
+                                                             :indexer 'crux.lucene.extension-test/->custom-eav-indexer}})]
+    (submit+await-tx node [[:crux.tx/put {:crux.db/id 0, :description "Some Entity"}]
+                           [:crux.tx/put {:crux.db/id 1, :description "Another entity"}]])
+
+    (let [db (c/db node)]
+      ;; text-search + leading-wildcard
+      (t/is (= [[0 "Some Entity" 2.0]
+                [1 "Another entity" 1.0]]
+               (let [a :description]
+                 (custom-text-search node db a (build-custom-query (->per-field-analyzer nil) (l/keyword->k a) "so* *En*")))))
```

Comment:
That the user has to pass the analyzer here is a bit of a smell, I wonder if there's a way to get hold of the analyzer from the node?

Doesn't have to be one for now :)

Reply:
Yeah...that's exactly why `custom-text-search` is just example code for the time being :)